### PR TITLE
Handing default security, handling image & kaniko job name, Adding sample scenarios, Deleting jobs

### PR DESCRIPTION
**Pull request by Shehanir**
Image name and kaniko job name should be handle as follow. 
    When adding new API,
    - Image name = image_name : tag
    - kaniko job name = api_name + "kaniko"
    When updating API,
    - Image name = image_name : tag + "-" + update_timestamp
    - kaniko job name = api_name + "kaniko" + "-" + update_timestamp

> Adding sample scenarios to test security flows (basic, JWT and Oauth)

> Deleting jobs after completion to prevent remaining jobs and relevant artifacts in the cluster.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: apim-operator/pkg/controller/api/api_controller.go [hunk 1]
```diff
@@ -383,49 +386,72 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+			log.Info("use default security")
+			//use default security
+			//copy default sec in wso2-system to user namespace
+			securityDefault := &wso2v1alpha1.Security{}
+			//check default security already exist in user namespace
+			errGetSec := r.client.Get(context.TODO(), types.NamespacedName{Name: defaultSecurity, Namespace: userNameSpace}, securityDefault)
+
+			if errGetSec != nil && errors.IsNotFound(errGetSec) {
+				log.Info("default security not found in " + userNameSpace + " namespace")
+				log.Info("retrieve default-security from wso2-system namespace")
```
Comment (pubudu538) [inline apim-operator/pkg/controller/api/api_controller.go]: wso2-system could change!

Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go]: added a constant value

File: apim-operator/deploy/scenarios/scenario-2/basic-security.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# under the License.
+
+apiVersion: wso2.com/v1alpha1
+kind: Security
+metadata:
+  name: petstorebasic
+spec:
+  # Security - Basic
+  type: basic
+  credentials: secret-basic
```
Comment (AmaliMatharaarachchi) [inline apim-operator/deploy/scenarios/scenario-2/basic-security.yaml]: It's better to add a new line EOF

Comment (Shehanir) [reply apim-operator/deploy/scenarios/scenario-2/basic-security.yaml:24]: Added

**Pull request merged**